src/main/java/com/mraof/minestuck/effects/SuspicionEffect.java

```diff
@@ -0,0 +1,136 @@
+package com.mraof.minestuck.effects;
+
+import com.mraof.minestuck.Minestuck;
+import net.minecraft.commands.arguments.CompoundTagArgument;
+import net.minecraft.world.damagesource.DamageSource;
+import net.minecraft.world.effect.MobEffect;
+import net.minecraft.world.effect.MobEffectCategory;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.FlyingMob;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.animal.FlyingAnimal;
+import net.minecraft.world.entity.decoration.ArmorStand;
+import net.minecraftforge.event.entity.EntityMountEvent;
+import net.minecraftforge.event.entity.living.ZombieEvent;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.common.Mod;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * The Suspicion effect causes mobs to forcibly move away from other entities.
+ * @see #onMount(net.minecraftforge.event.entity.EntityMountEvent) Cancels out attempts to ride affected entities
+ * @see #summonAttempt(ZombieEvent.SummonAidEvent) Cancels out attempts to spawn reinforcements
+ */
+@Mod.EventBusSubscriber(modid = Minestuck.MOD_ID, bus = Mod.EventBusSubscriber.Bus.FORGE)
+public class SuspicionEffect extends MobEffect
+{
+	/**
+	 * Used to fire some functionality in {@link #applyEffectTick(LivingEntity, int)} only once.
+	 */
+	private boolean hasInitialized = false;
+	/**
+	 * Used to enable the effect at all, blocking the {@link #applyEffectTick(LivingEntity, int)} and {@link #onMount(EntityMountEvent)} functions if false.
+	 * It is true if the afflicted entity is not an {@link net.minecraft.world.entity.decoration.ArmorStand} instance and doesn't have the "NoAI" NBT.
+	 */
+	private boolean isPushable = true;
+	/**
+	 * Used to enable the pushing effect on flying entities, as an alternative for the {@link LivingEntity#isOnGround()} check.
+	 */
+	private boolean isFlying = false;
```

Comment:
Instances of `MobEffect` are global/akin to a singleton object, and NOT specific to a specific entity. You should therefore not use entity-specific info in fields here as the same effect and fields will be used for ALL entities. Try instead defining reusable functions that figure out if a given entity is pushable or is flying.

Reply:
Ah, the usage of "MobEffectInstance" threw me off. Fixed, but I'm concerned that now there might be too many checks per tick. It depends on the performance impact.